CMakeLists.txt

```diff
@@ -557,23 +519,11 @@ install(FILES ${SERIALBOX_CXX_CONFIG_FILE}
         DESTINATION ${CMAKE_INSTALL_PREFIX}/include/serialbox/core)
 include_directories(${CMAKE_BINARY_DIR}/src)
 
-# Setup include directories and glob the headers (for IDEs)
-set(SERIALBOX_INCLUDE_DIRS ${CMAKE_CURRENT_SOURCE_DIR}/src)
-include_directories(${SERIALBOX_INCLUDE_DIRS})
-
-file(GLOB_RECURSE SERIALBOX_HEADERS 

```

Comment:
This is another CMake anti-pattern. We should avoid `file(GLOB ` unless we have a good reason for. 
In particular this one is collecting all the headers of the library for multiple uses: 
Here 
https://github.com/eth-cscs/serialbox2/pull/167/files#diff-15547c54d3d4898a882b4ab7b3cee381L34
it is used to pass all the heades to the target of a compiled example, irrespective of whether the headers belong to that target or not. 
Instead what we should do is always specify the headers (that belong to the target) as part of the sources being passed to that target: 
https://github.com/eth-cscs/serialbox2/pull/167/files#diff-de9c0f285f36311038c1222eb1daa005R35
Adding the headers to a target is required, because otherwise they are not displayed in the IDE project. 
So always add the corresponding headers to the sources of a library/executable target. 
If for some reasons, some headers/sources do not belong to any target, you can still used the following yoda function in order to make them visible in your IDE
https://github.com/MeteoSwiss-APN/yoda/blob/master/cmake/yodaAddCustomDummyTarget.cmake